### PR TITLE
wip-asdpc-number-separators: submit draft

### PR DESCRIPTION
**Pull request by aesedepece**

### CODE REVIEW

File: wip-asdpc-number-separators.md [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+## Abstract
+
+This introduces additional rules when processing in Witnet's data retrieval and aggregation engine (RAD), such that
+APIs that provide numbers formatted in a non-conventional manner can also be used as data sources.
+
+## Motivation and rationale
+
+Witnet is designed as a generalist oracle: there is no need for the nodes to select and publish specific datasets
+that they want to offer to smart contracts. On the contrary, it is rather on the data consumers (the smart contract
+developers) to explicitly select multiple data sources to be queried over HTTPS or orther protocols, define how to
```
Comment (Tommytrg) [inline wip-asdpc-number-separators.md:23]:
```suggestion
developers) to explicitly select multiple data sources to be queried over HTTPS or other protocols, define how to
```

File: wip-asdpc-number-separators.md [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+process the responses, and how to filter and aggregate the output of those multiple sources. In the Witnet protocol,
+we often call that specification _data request_, or more recently, _oracle query_. 
+
+This generalist approach has been proven more than effective to create decentralization at the data source level, as
+it enables the user to pick virtually any publicly accessible APIs as their data sources, while mitigating trust in any 
+single of those by means of basic filtering and aggregation techniques.
+
+This approach comes however with a few challenges, as public APIs come in all sizes and colors, and each of them will
```
Comment (Tommytrg) [inline wip-asdpc-number-separators.md:31]: Is this easier to follow?
```suggestion
However, this approach comes with a few challenges, as public APIs come in all sizes and colors, and each of them will
```

File: wip-asdpc-number-separators.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+arguments to any number processing RAD operators.
+
+## Specification
+
+### Definitions
+
+- For the scope of this proposal, the _number operators_ are `Array::get_float`, `Array::get_integer`, `Map::get_float`,
+`Map::get_integer`, `String::to_float`, and `String::to_integer`.
+
+- A _decimal separator_ is the character or sequence of characters that mark the separation between the integer part and
```
Comment (Tommytrg) [inline wip-asdpc-number-separators.md:67]:
```suggestion
- A _decimal separator_ is the character or sequence of characters that marks the separation between the integer part and
```

File: wip-asdpc-number-separators.md [hunk 4]
```diff
@@ -0,0 +1,151 @@
...
+### Pre-processing of separators in number operators
+
+**(6)** All number operators MUST pre-process numbers in such a way that any occurrence of the user-provided thousands 
+separator or the default thousands separator as defined in (3) are removed from the number (or equivalently, replaced
+by an empty string).
+
+**(7)** All number operators MUST pre-process numbers in such a way that any occurrence of the user-provided decimal
+separator are replaced by the default decimal separator as defined in (4).
+
+**(8)** Replacement of decimal separators do not affect processing of integer numbers, and hence the decimal separator
```
Comment (Tommytrg) [inline wip-asdpc-number-separators.md:98]:
```suggestion
**(8)** Replacement of decimal separators does not affect the processing of integer numbers, and hence the decimal separator 
```

File: wip-asdpc-number-separators.md [hunk 5]
```diff
@@ -0,0 +1,151 @@
...
+those coming from implementers, that is, the new rules.
+- Implementers MUST apply the proposed logic when resolving oracle queries.
+
+As a result:
+
+- Commitments and reveals signed by non-implementers MAY differ from those signed by implementers.
+- Oracle queries MAY resolve to `InsufficientConsensus` given a witnessing committee formed by a mixed sample of
+implementers and non-implementers.
+
+Due to this last points, this MUST be considered a consensus-critical protocol improvement. An adoption plan MUST be
```
Comment (Tommytrg) [inline wip-asdpc-number-separators.md:122]:
```suggestion
Due to these last points, this MUST be considered a consensus-critical protocol improvement. An adoption plan MUST be
```

**Comment by aesedepece** [comment]
With no opposing comments, this proposal meets all the WIP-0001 requirements to be published as a draft. I'll merge it along #88 and #89

**Comment by aesedepece** [comment]
This is now published as https://github.com/witnet/WIPs/blob/master/wip-0024.md

**Pull request closed without merging**